### ISSUE TITLE
New Features Introduction

### ISSUE THREAD

**Issue by ACG8** [sIKE23/Mage-Wars#181]
The new release has so many new features, players might be overwhelmed/not notice them all. I thought it might be good to put together a small tutorial that introduces them to the new features. The parameters are:

- It would be an askChoice function where each button gives information about a specific feature

- It will appear automatically the first time the player opens a game with the new version (and ONLY then), and afterwards will be available via a menu command

- It will make use of the textFunctions script to store all the tutorial text in a .txt file, which we can modify for future releases to explain functions for those

- Basically, it is a sort of in-game interactive changelog explaining how to use the latest game features

I'm not sure what the best thing to name it would be, though.

**Comment by sIKE23** [comment]
Sounds good we can work out a name  for as we go, take a look at the
initializeGame() function and the changelog.py, because I there is
overlapping functionality there.....

//FC

**Comment by ACG8** [comment]
Okay, great! I never really understood what changelog does, but if it triggers only when the player is new to the current version, it sounds like the perfect place to trigger this function. I'm still not sure how it works; do you think you can integrate the new function into the existing system for notifying players about new features?

**Comment by sIKE23** [comment]
I grabbed it from Brines MTG code because people kept complaining that
there was no documentation. All it is doing is looking at version
numbers and popping up a screen letting them know what documentation is
available. You should be able to put in your code in either
the initializeGame() function to bring up whatever text you want or create
your own function and call it from there......or you can completely replace
it but continue to use the Setting "lastVersion".......

//FC

**Comment by sIKE23** [comment]
Can we get a button to bring up the changelog? I don't think we need to put all of the changes into this features file, but we should be able to link off of it .....

**Comment by ACG8** [comment]
Sure. Where is the changelog? I see the notification that pops up when you first install a new version, but how do you get the actual changelog?

**Comment by sIKE23** [comment]
It is an html file in the docs folder.....

//FC

**Comment by ACG8** [comment]
Okay, I don't see any commands to open up HTML documents. You can access it in the documents tab, but there doesn't seem to be a function that does the same. Perhaps this is like loading decks (in the sense that the OCTGN engine can do it, but not the module)? Or do you know of a function that will open an html file?

**Comment by sIKE23** [comment]
Yeah I don't see anything either though it "should" be exposed to
Python.....

**Comment by sIKE23** [comment]
Yeah there is not....

**Comment by sIKE23** [comment]
Schwenkgott's thoughts on this....

![new feature menu](https://cloud.githubusercontent.com/assets/5695175/6673634/aacc0bca-cbe2-11e4-990b-1e16a65389f7.jpg)

I can see making the button text simpler, what about splitting Searchable Codex and Rulings/Clarifications into two buttons? Your thoughts?

**Comment by sIKE23** [comment]
The other issue, is if you are not the card controller you do not get the Rulings/Clarifications context menu item. I am not sure if you can do anything about it though.

**Comment by ACG8** [comment]
Oh, that is a shame. You are right though; I don't think I can do anything about that. I could at least give players a way to search for rulings/clarifications by name.

I think that is a good point about button text. I can see about simplifying it (or you could, of course; maybe you have a better idea of what to put),

**Comment by ACG8** [comment]
We need to fill out the New Features explanation for the upcoming release.
